presto-main/src/main/java/com/facebook/presto/operator/aggregation/QuantileDigestAggregationFunction.java

```diff
@@ -13,215 +13,72 @@
-        final MethodHandle inputFunction;
-        switch (valueType.getDisplayName()) {
-            case StandardTypes.DOUBLE:
-                inputFunction = INPUT_DOUBLE;
-                break;
-            case StandardTypes.REAL:
-                inputFunction = INPUT_REAL;
-                break;
-            case StandardTypes.BIGINT:
-                inputFunction = INPUT_BIGINT;
-                break;
-            default:
-                throw new IllegalArgumentException(format("Unsupported type %s supplied", valueType.getDisplayName()));
-        }
-
-        switch (arity) {
-            case 1:
-                // weight and accuracy unspecified
-                return insertArguments(inputFunction, 2, DEFAULT_WEIGHT, DEFAULT_ACCURACY);
-            case 2:
-                // weight specified, accuracy unspecified
-                return insertArguments(inputFunction, 3, DEFAULT_ACCURACY);
-            case 3:
-                // weight and accuracy specified
-                return inputFunction;
-            default:
-                throw new IllegalArgumentException(format("Unsupported number of arguments: %s", arity));
-        }
+        return "Returns a qdigest from the set of reals, bigints, doubles";
     }
 
-    private static List<ParameterMetadata> createInputParameterMetadata(List<Type> valueTypes)
+    public static void inputDouble(StatisticalDigestState state, double value, long weight, double parameter)
     {
-        return ImmutableList.<ParameterMetadata>builder()
-                .add(new ParameterMetadata(STATE))
-                .addAll(valueTypes.stream().map(valueType -> new ParameterMetadata(INPUT_CHANNEL, valueType)).collect(Collectors.toList()))
-                .build();
+        inputBigint(state, doubleToSortableLong(value), weight, parameter);
     }
 
-    public static void inputDouble(QuantileDigestState state, double value, long weight, double accuracy)
-    {
-        inputBigint(state, doubleToSortableLong(value), weight, accuracy);
-    }
-
-    public static void inputReal(QuantileDigestState state, long value, long weight, double accuracy)
+    public static void inputReal(StatisticalDigestState state, long value, long weight, double accuracy)
     {
         inputBigint(state, floatToSortableInt(intBitsToFloat((int) value)), weight, accuracy);
     }
 
-    public static void inputBigint(QuantileDigestState state, long value, long weight, double accuracy)
+    public static void inputBigint(StatisticalDigestState state, long value, long weight, double accuracy)
     {
-        QuantileDigest qdigest = getOrCreateQuantileDigest(state, verifyAccuracy(accuracy));
-        state.addMemoryUsage(-qdigest.estimatedInMemorySizeInBytes());
-        qdigest.add(value, verifyWeight(weight));
-        state.addMemoryUsage(qdigest.estimatedInMemorySizeInBytes());
+        checkArgument(accuracy > 0 && accuracy < 1, "Percentile accuracy must be exclusively between 0 and 1, was %s", accuracy);
```

Comment:
These logic used to be in `QuantileDigestFunctions`. I don't have strong preference as where they should sit or whether they should be a separate function or not. @tdcmeehan since you implemented it first, what do you think? Is this a deliberate change?

Reply:
Seems like it should be in a helper.  By the way, we should make sure a PrestoException is thrown so it isn't reported as a system exception when an out of bounds quantile is requested.  This may have been a bug in the prior implementation.